src/gefs_6h_ave_1mem.fd/gefs_6h_ave_1mem.f90

```diff
@@ -0,0 +1,538 @@
+!      data pdt%iprm/ 192,193,192,193,193,193,17,18,11,10, 7,196, 1,  1,  1,  5,  4,  3,193,194,195,194,195,  5,  4,  0,  0,  0,  0,  0,  0,  0,  0,  0,193,16,42,  5, 8, 10, 9/!ParameterNumberbyProductDiscipline and Parameter Category
+!      data pdt%iffs/   1,  1,  1,  1,  8,  8, 1, 1, 1, 1, 1,  1, 1,200,211,234,224,214,  1,  1,  1,  1,  1,103,103,233,232,233,223,222,223,213,212,213,200, 1, 1,  1,  1,  1,  1/
+!      integer,dimension(nfield)::disc
+!      data disc/0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0/
+
+
+      data pdt%npdt/   8,  8,  8,  8,  8,  8, 8, 8, 8, 8, 8,  8, 8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8,  8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8, 8/
+      data pdt%icat/   5,  5,  4,  4,  4,  5, 2, 2, 0, 0, 1,  1,19,  6,  6,  6,  6,  6,  0,  3,  3,  4,  4,  0,  0,  3,  3,  0,  3,  3,  0,  3,  3,  0,  6, 0, 1,  0, 1,  1, 1, 1/!Parameter Category by Product Discipline
+      data pdt%iprm/ 192,193,192,193,193,193,17,18,11,10, 7,196, 1,  1,  1,  5,  4,  3,193,194,195,194,195,  5,  4,  0,  0,  0,  0,  0,  0,  0,  0,  0,193,16,42,  5, 50, 8, 10, 9/!ParameterNumberbyProductDiscipline and Parameter Category
+      data pdt%iffs/   1,  1,  1,  1,  8,  8, 1, 1, 1, 1, 1,  1, 1,200,211,234,224,214,  1,  1,  1,  1,  1,103,103,233,232,233,223,222,223,213,212,213,200, 1, 1,  1, 1, 1,  1,  1/
+      integer,dimension(nfield)::disc
+      data disc/0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,0,0,0,0/
+
+
+      integer :: n_time
+      parameter(n_time=3) !from fhr 00 to 06
+      integer :: j,jpdtn,jgdtn
+      integer,dimension(200) :: jids,jpdt,jgdt
+      logical :: unpack=.true.
+      integer :: jdisc      ! discipline#, table 0.0(met:0 hydro:1 land:2)
+      integer :: day,month,year,hour,fhour
+      character(len=256) :: datafile(n_time),outfile,file_dir,out_dir,outfile03,outfile006
+      character(len=8) :: file_date
+!      character(len=5) :: ens_mem
+      integer :: unit, ifid,ifid1,nx,ny,iret,jret,i,k,ifh,nfi,maxgrd,ifd,ind,ens_id
+      character(len=3) ::sfh
+      character(len=2) ::sdy
+      character(len=2) ::smonth
+      character(len=4) ::syr
+      character(len=8) :: pabbrev
+      character(len=30) :: labbrev
+      character(len=30) :: labbrev_short
+
+      real, allocatable :: var_save(:,:) ! (maxgrd,nfi)
+      real, allocatable :: var_save_acc(:) ! (maxgrd)
+      real, allocatable :: var_save_DSWRF(:) ! (maxgrd)
+      real, allocatable :: var_save_USWRF(:) ! (maxgrd)
+      real, allocatable :: apcp_6h(:) ! (maxgrd)
+      real, allocatable :: acpcp_6h(:) ! (maxgrd)
+      real, allocatable :: apcp_3h(:) ! (maxgrd)
+      real, allocatable :: acpcp_3h(:) ! (maxgrd)
+      real, allocatable :: dpcp(:) ! (maxgrd)
+
+
+      call GET_COMMAND_ARGUMENT(1, file_date)
+      call get_environment_variable("file_dir", file_dir)
+      call get_environment_variable("out_dir", out_dir)
```

Comment:
please lets not use environment variables.  Either use a namelist or command-line argument.